fieldsolver/gridGlue.cpp

```diff
@@ -92,6 +92,118 @@ template <typename T, int stencil> void computeCoupling(dccrg::Dccrg<SpatialCell
+*/
+void filterMoments(dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>& mpiGrid,
+                           FsGrid< std::array<Real, fsgrids::moments::N_MOMENTS>, FS_STENCIL_WIDTH> & momentsGrid,
+                           FsGrid< fsgrids::technical, FS_STENCIL_WIDTH> & technicalGrid) 
+{
+
+
+
+   // Kernel Characteristics
+   const int stencilWidth = 5;   // dimension size of a 3D kernel
+   const int kernelOffset = 2;   // offset of 5 pointstencil 3D kernel => (floor(stencilWidth/2);)
+   const Real kernelSum=729.0;
+   const Real kernel[5][5][5] ={
+                                 {{ 1,  2,  3,  2,  1},
+                                 { 2,  4,  6,  4,  2},
+                                 { 3,  6,  9,  6,  3},
+                                 { 2,  4,  6,  4,  2},
+                                 { 1,  2,  3,  2,  1}},
+
+                                 {{ 2,  4,  6,  4,  2},
+                                 { 4,  8, 12,  8,  4},
+                                 { 6, 12, 18, 12,  6},
+                                 { 4,  8, 12,  8,  4},
+                                 { 2,  4,  6,  4,  2}},
+
+                                 {{ 3,  6,  9,  6,  3},
+                                 { 6, 12, 18, 12,  6},
+                                 { 9, 18, 27, 18,  9},
+                                 { 6, 12, 18, 12,  6},
+                                 { 3,  6,  9,  6,  3}},
+
+                                 {{ 2,  4,  6,  4,  2},
+                                 { 4,  8, 12,  8,  4},
+                                 { 6, 12, 18, 12,  6},
+                                 { 4,  8, 12,  8,  4},
+                                 { 2,  4,  6,  4,  2}},
+
+                                 {{ 1,  2,  3,  2,  1},
+                                 { 2,  4,  6,  4,  2},
+                                 { 3,  6,  9,  6,  3},
+                                 { 2,  4,  6,  4,  2},
+                                 { 1,  2,  3,  2,  1}}
+                                 };
+
+   // Update momentsGrid Ghost Cells
+   momentsGrid.updateGhostCells(); 
+
+
+   // Get size of local domain and create swapGrid for filtering
+   const int *mntDims= &momentsGrid.getLocalSize()[0];  
+   const int maxRefLevel = mpiGrid.mapping.get_maximum_refinement_level();
+   FsGrid<std::array<Real, fsgrids::moments::N_MOMENTS>, FS_STENCIL_WIDTH> swapGrid = momentsGrid;  //swap array 
+
+   // Filtering Loop
+   for (int blurPass = 0; blurPass < Parameters::maxFilteringPasses; blurPass++){
+
+      // Blurring Pass
+      #pragma omp parallel for collapse(2)
+      for (int k = 0; k < mntDims[2]; k++){
+         for (int j = 0; j < mntDims[1]; j++){
+            for (int i = 0; i < mntDims[0]; i++){
+
+               //  Get refLevel level
+               int refLevel = technicalGrid.get(i, j, k)->refLevel;
+
+               //  Get refLevel level
+               // Skip pass
+               if (blurPass >= P::numPasses[refLevel] ||
```

Comment:
You changed P::numPasses into a map instead of a vector - now we're doing a map search for every cell for every refinement pass. I think this is probably not a fast approach and would suggest returning to a vector, unless there's a good reason?